Spr23

Gera-sim opened the pull request.

Gera-sim committed:
SPR23 Legacy ico

Gera-sim committed:
SPR23 - not from goals: firebase analytics

Gera-sim committed:
SPR23 - Версия с NPE

Gera-sim committed:
SPR23 - ready

Gera-sim committed:
SPR23 - nav_graph

Charmandik commented on app/src/main/java/com/example/playlistmaker/medialibrary/ui/playlists/AddPlayListViewModel.kt:
```diff
@@ -1,26 +1,54 @@
+    }
 
+    companion object {
+        private const val CLICK_DEBOUNCE_DELAY = 1000L
```
При именовании констант для промежутка времени рекомендую всегда в конце добавлять единицу времени, например _MILLIS, _IN_SECONDS, _HOURS и т.д.

Gera-sim replied:
Это всё спешка (((

Charmandik commented on .idea/deploymentTargetDropDown.xml:
Добавь этот тип файлов в гитигнор, их не нужно отслеживать в гите

Charmandik commented on app/src/main/java/com/example/playlistmaker/medialibrary/ui/playlists/AddPlaylistFragment.kt:
```diff
@@ -13,39 +11,61 @@ import android.widget.Toast
 
     private val addPlayListViewModel by viewModel<AddPlayListViewModel>()
 
     private lateinit var confirmDialog: MaterialAlertDialogBuilder
```
Вместо использования lateinit, я рекомендую тебе использовать подход с добавлением nullability к типу объекта с оператором ?. и сразу проинициализировать его значением null. Это решение позволяет избежать ошибок, связанных с попытками доступа к неинициализированным объектам, которые могут привести к непредсказуемому поведению или крашам приложения. Хотя это может добавить некоторую дополнительную работу при обращении к объекту через оператор ?. , но это значительно повысит безопасность приложения, так как ты будешь уверен, что объект всегда либо имеет значение, либо равен null, и не будет вызывать исключительных ситуаций.Можешь еще прочитать статью на эту тему https://habr.com/ru/companies/maxilect/articles/590061

Gera-sim replied:
lateinit был же до исправления. сейчас:
    private var _binding: FragmentAddplaylistBinding? = null
    private val binding get() = _binding!!
    + onDestroyView для защиты от утечек

Charmandik commented on app/src/main/java/com/example/playlistmaker/main/MainActivity.kt:
```diff
@@ -29,13 +30,21 @@ class MainActivity : AppCompatActivity() {
+                    if (binding.navigationView.visibility == View.VISIBLE) {
+                        binding.navigationView.animation = AnimationUtils.loadAnimation(this, R.anim.slide_out_down)
+                        binding.navigationView.animate()
+                        binding.navigationView.visibility = View.GONE
```
Для управления видимостью можно использовать свойство isVisible, устанавливая значение false для View.GONE и true для View.VISIBLE
Например, если нам нужно установить View.VISIBLE:
myViewName.isVisible = true
Или если нужно установить View.GONE, то : 
myViewName.isVisible = false

Возможно, тебе понравится такой подход

Gera-sim replied:
Предложенный подход используется в других вьюхах. Тут осталось пока так - для разнообразия  ))

Gera-sim replied:
Спасибо!

Charmandik commented on app/src/main/java/com/example/playlistmaker/medialibrary/data/PlayListsRepositoryImpl.kt:
```diff
@@ -1,30 +1,89 @@
             trackPlayListEntity = TrackPlayListEntity(null, playListId, track.trackId)
         )
 
+    override suspend fun getPlayList(playListId: Int): PlayList =
```
В Kotlin лучше использовать "return" вместо "=", потому что это помогает делать код более читаемым и явным. Когда ты используешь "return", это явно указывает, что функция возвращает определенное значение. В то же время, использование "=" может вызывать путаницу, особенно если в функции есть несколько выражений.

Gera-sim committed:
SPR23 - правки ревью:
❌Если пользователь находится на экране «Плейлист» и видит список треков, то при длительном нажатии (onLongClick) на элемент списка, пользователь видит диалог с заголовком «Удалить трек», сообщением «Вы уверены, что хотите удалить трек из плейлиста?» и кнопками «Нет» и «Да». (текст из ТЗ)

- Исправил выводимые сообщения удаления трека и плейлиста.

❌В неактивном состоянии кнопка находится, если поле ввода названия плейлиста пустое (пользователь стёр текущее название).

- Ошибку не удалось воспроизвести: кнопка Сохранить корректно обрабатывается на различных виртуальных и физических устройствах.

Gera-sim committed:
SPR23 - мелкие правки

Gera-sim committed:
SPR23 - bugFix

Gera-sim commented:
Прошу обратить внимание: в тексте задания и в чеклисте под ним некоторые фразы и вопросы для всплывающих окон и их кнопок и т.д. не совпадают.

Gera-sim committed:
SPR23 - bugFix2
